pkg/api/util/ipc_endpoint_test.go

```diff
@@ -0,0 +1,212 @@
+
+	return conf
+}
+
+func TestNewIPCEndpoint(t *testing.T) {
+	conf := pkgconfigmodel.NewConfig("datadog", "DD", strings.NewReplacer(".", "_"))
+
+	// create a fake auth token
+	authTokenFile, err := os.CreateTemp("", "")
+	assert.NoError(t, err)
+	authTokenPath := authTokenFile.Name()
+	os.WriteFile(authTokenPath, []byte("0123456789abcdef0123456789abcdef"), 0640)
+
+	// set minimal configuration that IPCEndpoint needs
+	conf.Set("auth_token_file_path", authTokenPath, pkgconfigmodel.SourceAgentRuntime)
+	conf.Set("cmd_host", "localhost", pkgconfigmodel.SourceAgentRuntime)
+	conf.Set("cmd_port", "6789", pkgconfigmodel.SourceAgentRuntime)
+
+	// test the endpoint construction
+	end, err := NewIPCEndpoint(conf, "test/api")
+	assert.NoError(t, err)
+	assert.Equal(t, end.target.String(), "https://localhost:6789/test/api")
+}
+
+func TestNewIPCEndpointWithCloseConnection(t *testing.T) {
+	conf := pkgconfigmodel.NewConfig("datadog", "DD", strings.NewReplacer(".", "_"))
+
+	// create a fake auth token
+	authTokenFile, err := os.CreateTemp("", "")
+	assert.NoError(t, err)
+	authTokenPath := authTokenFile.Name()
+	os.WriteFile(authTokenPath, []byte("0123456789abcdef0123456789abcdef"), 0640)
+
+	// set minimal configuration that IPCEndpoint needs
+	conf.Set("auth_token_file_path", authTokenPath, pkgconfigmodel.SourceAgentRuntime)
+	conf.Set("cmd_host", "localhost", pkgconfigmodel.SourceAgentRuntime)
+	conf.Set("cmd_port", "6789", pkgconfigmodel.SourceAgentRuntime)
+
+	// test constructing with the CloseConnection option
+	end, err := NewIPCEndpoint(conf, "test/api", WithCloseConnection(true))
+	assert.NoError(t, err)
+	assert.True(t, end.closeConn)
+}
+
+func TestIPCEndpointGet(t *testing.T) {
+	gotURL := ""
+	ts := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		gotURL = r.URL.String()
+		_, _ = io.ReadAll(r.Body)
+		w.Write([]byte("ok"))
+	}))
+	defer ts.Close()
+
+	conf := createConfig(t, ts)
+	end, err := NewIPCEndpoint(conf, "test/api")
+	assert.NoError(t, err)
+
+	// test that DoGet will hit the endpoint url
+	res, err := end.DoGet()
+	assert.NoError(t, err)
+	assert.Equal(t, res, []byte("ok"))
+	assert.Equal(t, gotURL, "/test/api")
+}
+
+func TestIPCEndpointGetWithValues(t *testing.T) {
+	gotURL := ""
+	ts := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		gotURL = r.URL.String()
+		_, _ = io.ReadAll(r.Body)
+		w.Write([]byte("ok"))
+	}))
+	defer ts.Close()
+
+	conf := createConfig(t, ts)
+	// set url values for GET request
+	v := url.Values{}
+	v.Set("verbose", "true")
+
+	// test construction with option for url.Values
+	end, err := NewIPCEndpoint(conf, "test/api")
+	assert.NoError(t, err)
+
+	// test that DoGet will use query parameters from the url.Values
+	res, err := end.DoGet(WithValues(v))
+	assert.NoError(t, err)
+	assert.Equal(t, res, []byte("ok"))
+	assert.Equal(t, gotURL, "/test/api?verbose=true")
+}
+
```

Comment:
It seems that we are not testing the `WithHTTPClient` option pattern

Reply:
I'm honestly not sure how to test that option, as I included it simplify for completion's sake. I'm open to suggestions, and also I'll think on it myself.

Reply:
Now testing that an http endpoint can be used by creating a non-https client and passing it to `WithHTTPClient`.